R/IOHproblem.R

```diff
@@ -0,0 +1,79 @@
+#' S3 print function for IOHproblem
+#'
+#' @param x The IOHproblem to print
+#' @param ... Arguments for underlying function
+#'
+#' @export
+print.IOHproblem <- function(x, ...) {
+  cat(as.character.IOHproblem(x, ...))
+}
+
+#' S3 generic cat function for IOHproblem
+#'
+#' @param x The IOHproblem to print
+#'
+#' @export
+cat.IOHproblem <- function(x) cat(as.character(x))
+
+#' S3 generic as.character function for IOHproblem
+#'
+#' @param x The IOHproblem to print
+#' @param ... Arguments for underlying function
+#'
+#' @export
+as.character.IOHproblem <- function(x, ...) {
+  sprintf('IOHproblem (Instance %d of function %d %dD)', x$instance, x$function_id, x$dimension)
+}
+
+#' Get the next function of the currently initialized suite
+#'
+#' @param experimenter The IOHexperimenter object
+#'
+#' @return An IOHproblem object
+#' @export
+next_problem <- function(experimenter) {
+  ans <- c_get_next_problem()
+  if (is.null(ans) || is.null(ans$problem)) return(NULL)
+
+  return(structure(
+    list(
+      dimension = ans$dimension,
+      function_id = ans$problem,
+      instance = ans$instance,
+      fopt = c_get_fopt(),
+      xopt = c_get_xopt(),
+      params.track = experimenter$param.track,
+      obj_func = function(x) {
+        stopifnot( all(x %in% c(0,1) ))
+        if (is.null(dim(x))) x <- t(x)
+        if (ncol(x) != ans$dimension) x <- t(x)
+
+        stopifnot(ncol(x) == ans$dimension)
+        apply(x, 1, c_eval)
+      },
+      target_hit = function() {
+        c_is_target_hit()
+      },
+      set_parameters = function(param_vals){
+        stopifnot( length(param_vals) == length(experimenter$param.track) )
+        c_set_parameters(param_vals)
+      }
+
+    ),
+    class = c('IOHproblem', 'list')
+  ))
+}
+
+#' Reset the IOHproblem
+#'
+#' @param problem The IOHproblem object
+#'
+#' @return An IOHproblem object
+#' @export
+reset_problem <- function(problem) {
+  ans <- c_reset_problem()
```

Comment:
Please check this with Furong: after calling c_reset_problem, what would happen to the output file? a new line in .info file? a new *.dat file?

Reply:
This reset_problem creates a new header-line in the *.dat-file and a new 'value' in the .info file. So until you go to the next function, you keep using the same .dat and .info files

Reply:
Good, clear to me now!